deps/chakrashim/core/lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -506,7 +506,13 @@ bool JsrtDebuggerStackFrame::Evaluate(Js::ScriptContext* scriptContext, const ch
             resolvedObject.typeId = Js::JavascriptOperators::GetTypeId(resolvedObject.obj);
             JsrtDebuggerObjectBase::CreateDebuggerObject<JsrtDebuggerObjectProperty>(this->debuggerObjectsManager, resolvedObject, scriptContext, [&](Js::Var marshaledObj)
             {
-                *evalResult = (Js::DynamicObject*)marshaledObj;
+                Js::DynamicObject *debuggerObj = (Js::DynamicObject*)marshaledObj;
+                if (returnByValue)
+                {
+                    JsrtDebugUtils::AddPropertyToObject(debuggerObj, JsrtDebugPropertyId::value, resolvedObject.obj, scriptContext);
```

Comment:
based on our offline conversation- since JsrtDebuggerObjectBase::CreateDebuggerObject already sets the value property in some cases, it seems like it would be cleaner to just shovel this flag off to that function and let it take care of setting the function.
Also, I mildly prefer the name "forceSetValueProp" over returnByValue